snuba/datasets/discover.py

```diff
@@ -89,11 +90,16 @@ def __init__(
 
     def select_storage(
         self, query: Query, request_settings: RequestSettings
-    ) -> ReadableStorage:
+    ) -> Tuple[ReadableStorage, QueryTranslator]:
         table = detect_table(
             query, self.__abstract_events_columns, self.__abstract_transactions_columns,
         )
-        return self.__events_table if table == EVENTS else self.__transactions_table
+        translator = CopyTranslator()
```

Comment:
Obviously the translation will depend on the storage we select, and the selection itself may end up being the result of the translation (depending how we decide to couple translation and storage).